ambari-common/src/main/python/resource_management/libraries/execution_command/execution_command.py

```diff
@@ -231,10 +246,74 @@ def get_ambari_jdk_name(self):
   def need_refresh_topology(self):
     return self.__get_value('commandParams/refresh_topology', False)
 
+  def check_only_update_files(self):
+    return self.__get_value('commandParams/update_files_only', False)
+
 
   """
   Role related variables section
   """
 
   def is_upgrade_suspended(self):
-    return self.__get_value('roleParams/upgrade_suspended', False)
\ No newline at end of file
+    return self.__get_value('roleParams/upgrade_suspended', False)
+
+  """
+  Cluster Host Info
+  """
+
+  def get_namenode_hosts(self):
```

Comment:
No we cannot have component names in methods. The method needs to be generic. You can add special-casing within the method to handle inconsistencies. Ideally we should fix inconsistencies in the future. But the library interface MUST BE generic. Otherwise how will it work for third-party mpacks which we are not even aware off?

Example:

```
def get_component_hosts(self, component_name): 
  key = "clusterHostInfo/" + component_name + "_hosts"
  if component_name == "oozie_server":
    key = "clusterHostInfo/" + component_name
  return self.__get_value(key, [])
```

Reply:
Jayush, thanks. I thought wrongly when I implemented like this at that time. I just uploaded the  new patch.